### PR TITLE
Bugfix/aito 321 big integer data

### PR DESCRIPTION
**Pull request by araufuturice**
Several improvements to csv table inference to make it more lenient with odd CSVs:
- CSV is now lenient for tables with too many fields. Such tables happen easily, if excel sheet is converted to CSV. 
- Columns with integers outside 32 bit integer range are now converted to string
- Empty columns are now converted to string
- Columns containing only whitespace no more cause error in language detection. 
- Table schema is always inferred on dataframe conversions to manage too large numeric values.
- None / missing values are now maintained on conversions. Also, a conversion is always forced for all columns, when writing the data 
- Additional utility class DataSeriesProperties was added to help with conversions with too large integers
- read_csv now uses the Python engine instead of C engine, because C engine couldn't manage extra fields in a row as promised in pandas documentation

NOTE: The new conversions are likely slower, as Python engine is used, type inference is forced and all columns are always converted.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -45,6 +45,34 @@ Documentation can be generated by:
...
+
+The Python SDK tests depend on the ODBC db drivers and progres and mysql functionality.
+
+These drivers can be installed in linux with:
+
+```
+sudo apt install unixodbc-dev odbc-postgresql postgresql-client default-mysql-client
+pip install pyodbc
+```
+Check the `.circleci/config.yml` for more instructions on installing the mysql odbc unicode drivers.
```
Comment (cventus) [inline README.rst]: This documentation ends up at https://pypi.org/project/aitoai/ and I'm not sure if `.circleci/config.yml` is available from there. Still, it doesn't really matter, having this info available is necessary.

File: aito/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = "0.4.0"
+__version__ = "2020.11.11.15.2.7.dev0"
```
Comment (cventus) [inline aito/__init__.py]: What does this version mean? Is it part of the deployment process?

File: aito/schema.py [hunk 3]
```diff
@@ -866,8 +930,7 @@ def __init__(self):
         super().__init__('String')
 
     def to_python_type(self):
-        return str
-
+       return str
```
Comment (cventus) [inline aito/schema.py]: Incorrect indentation :open_mouth: It starts with one broken window...

```suggestion
        return str
```

File: scripts/deploy [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (cventus) [inline scripts/deploy:1]: Is `python3` generally the name of the executable?

File: tests/io/in/sample_invoice/invoice_aito_schema.json [hunk 5]
```diff
@@ -1,23 +1,23 @@
 {
-    "columns": {
-        "Remark": {
+     "columns": {
```
Comment (cventus) [inline tests/io/in/sample_invoice/invoice_aito_schema.json]: Top-level indentation changed from 4 to 5?

**Review by cventus — APPROVED** [review]
Looks good, minor questions below :point_down:

File: aito/__init__.py (hunk 2 above)
Comment (tonnitommi) [reply aito/__init__.py]: I think the python sdk numbering has been following major.minor.release versioning all the way from the beginning, so we should not invent a new way now. This version to me sounds like 0.4.1, and at least for bigger updates, there has been a release candidate versioned "0.4.1rc1" before promoting to a published version.

Comment (araufuturice) [reply aito/__init__.py]: This is a good catch. I tried to test the local/development deployment, and didn't realize that it rewrites the project version. 

0.4.1rc1 and 0.4.1 sounds sensible. I'll update the versioning, once this is merged. I think I also need to recheck the git versions / tags.

File: scripts/deploy (hunk 4 above)
Comment (araufuturice) [reply scripts/deploy:1]: In my Ubuntu, python is mapped to python2, and you need to be explicit about using python3. As a consequence, the project scripts broke on my machine.

File: README.rst (hunk 1 above)
Comment (araufuturice) [reply README.rst]: That is a good catch. I'm obviously missing a good mental image on the consequences of various changes.

Still, doesn't all other testing related code require you to pull the Git repo, which means that you can access the .circleci/config.yml that way.

Comment (araufuturice) [reply README.rst]: I did remove the circle CI reference and added a link to the MySQL instructions.

I have to say that the best documentation for making the tests work is in the cicleci config, but maybe referring to the file looks a bit unprofessional in a public repository.

**Pull request merged**